alibi_detect/utils/_random.py

```diff
@@ -0,0 +1,80 @@
+from contextlib import contextmanager
+import random
+import numpy as np
+import os
+import sys
+from alibi_detect.utils.frameworks import has_pytorch, has_tensorflow
+
+# Pointer to the alibi_detect module. Used to set a global seed variable.
+_module_pointer = sys.modules[__name__]
+setattr(_module_pointer, '_seed', None)
+
+if has_tensorflow:
+    import tensorflow as tf
+if has_pytorch:
+    import torch
+
+
+def set_seed(seed: int):
+    """
+    Sets the Python, NumPy, TensorFlow and PyTorch random seeds (if installed).
+
+    Parameters
+    ----------
+    seed
+        Value of the random seed to set.
+    """
+    seed = max(seed, 0)  # TODO: This is a fix to allow --randomly-seed=0 in setup.cfg. To be removed in future
+    setattr(_module_pointer, '_seed', seed)
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+    if has_tensorflow:
+        tf.random.set_seed(seed)
+    if has_pytorch:
+        torch.manual_seed(seed)
+
+
+def get_seed() -> int:
+    """
+    Gets the seed set by :func:`set_seed`.
+
+    Example
+    -------
+    >>> from alibi_detect.utils._random import set_seed, get_seed
+    >>> set_seed(42)
+    >>> get_seed()
+    42
+    """
+    if _module_pointer._seed is not None:
+        return _module_pointer._seed
+    else:
+        raise RuntimeError('`set_seed` must be called before `get_seed` can be called.')
```

Comment:
What's the reason for not accepting `None` as valid? Think I'm a bit lost in the motivation of these utilities again...

Reply:
WRT to motivation, the primary motivation of this PR now is to provide the `fixed_seed` context manager for use in tests. We want to be able to run detectors deterministically, i.e. 

```
with fixed_seed(42):
   detector1 = MMDDrift(X_ref, ...)
   preds1 = detector1.predict(x_h0)
```

But want to avoid setting global seeds, since these transfer into the next `pytest` test (I've checked this with a simple demo). This is quite an issue since in some tests we don't want the seeds fixed (see #250). If we move to use generators (e.g. `np.random.Generator`) in the future, and have `MMDDrift(x_ref, ..., random_seed=...)`, we can probably get rid of this entire `_random` module. 

The `set_seed` and `get_seed` methods might be helpful to a user in some way, but they're primarily for use with `fixed_seed` so I've kept the submodule private for now.

Does the above help or am I barking up the wrong tree? 🙂